soroban-env-host/src/fees.rs

```diff
@@ -238,18 +241,23 @@ pub fn compute_write_fee_per_1kb(
         .write_fee_1kb_bucket_list_high
         .saturating_sub(fee_config.write_fee_1kb_bucket_list_low)
         .clamp_fee();
-    let bucket_list_size_before_reaching_target =
-        bucket_list_size_bytes.min(fee_config.bucket_list_target_size_bytes);
-    // Convert multipliers to i128 to make sure we can handle large bucket list
-    // sizes.
-    let mut write_fee_per_1kb = num_integer::div_ceil(
-        (fee_rate_multiplier as i128)
-            .saturating_mul(bucket_list_size_before_reaching_target as i128),
-        fee_config.bucket_list_target_size_bytes as i128,
-    )
-    .clamp_fee();
-    write_fee_per_1kb = write_fee_per_1kb.saturating_add(fee_config.write_fee_1kb_bucket_list_low);
-    if bucket_list_size_bytes > fee_config.bucket_list_target_size_bytes {
+    let mut write_fee_per_1kb: i64;
+    if bucket_list_size_bytes < fee_config.bucket_list_target_size_bytes {
+        let bucket_list_size_before_reaching_target =
+            bucket_list_size_bytes.min(fee_config.bucket_list_target_size_bytes);
+        // Convert multipliers to i128 to make sure we can handle large bucket list
+        // sizes.
+        write_fee_per_1kb = num_integer::div_ceil(
+            (fee_rate_multiplier as i128)
+                .saturating_mul(bucket_list_size_before_reaching_target as i128),
+            fee_config.bucket_list_target_size_bytes as i128,
+        )
+        .clamp_fee();
+        // no clamp_fee here
+        write_fee_per_1kb =
+            write_fee_per_1kb.saturating_add(fee_config.write_fee_1kb_bucket_list_low);
+    } else {
+        write_fee_per_1kb = fee_config.write_fee_1kb_bucket_list_high;
```

Comment:
From offline discussions - 
Siddharth - "So if we're over the target size, we charge a fixed fee (write_fee_1kb_bucket_list_high) for the bytes up to bucket_list_target_size_bytes? The calculations below only deal with the bytes after the target size. Was this intentional?".

Reply:
Nico - "The CAP has this written in a fairly convoluted way, I agree.
before this change:
```
write_fee_per_1kb(s) = (writeFee1KBBucketListHigh - writeFee1KBBucketListLow)*min(s, bucketListTargetSizeBytes)/bucketListTargetSizeBytes +
  writeFee1KBBucketListLow +
  (if s > bucketListTargetSizeBytes,
      bucketListWriteFeeGrowthFactor*
      (writeFee1KBBucketListHigh - writeFee1KBBucketListLow)*
        (s-bucketListTargetSizeBytes)/bucketListTargetSizeBytes)
```

In plain English, we just have 2 curves (well 3):
* before target, it's a line that starts at `(0, write_fee_1kb_bucket_list_low)` and ends at `(bucket_list_target_size_bytes, write_fee_1kb_bucket_list_high)`
     * as we allow the line to intercept with a low (negative) `write_fee_1kb_bucket_list_low` we clamp it with `MINIMUM_WRITE_FEE_PER_1KB`  -> this creates a first "flat" portion to the first part of the fee curve.
* after target, it's the line that starts at `(bucket_list_target_size_bytes, write_fee_1kb_bucket_list_high)` with a slope of `bucketListWriteFeeGrowthFactor*
      (writeFee1KBBucketListHigh - writeFee1KBBucketListLow)/bucketListTargetSizeBytes`  --> `bucketListWriteFeeGrowthFactor` faster than the slope in step 1"

Reply:
@MonsieurNicolas am I right in saying that the main motivation for this change was to allow for the increase of the slope before the target size is hit without increasing writeFee1KBBucketListHigh? I'm trying to figure out why we can't just increase writeFee1KBBucketListHigh instead. The other important part was setting the min.